networkit/cpp/distance/Diameter.cpp

```diff
@@ -304,22 +298,21 @@ edgeweight Diameter::estimatedVertexDiameterPedantic(const Graph& G) {
         });
         vd ++; //we need the number of nodes, not the number of edges
     }
-    else {
+    else if (G.isEmpty()) {
+        vd = 0;
+    } else {
         ConnectedComponents cc(G);
         DEBUG("finding connected components");
         cc.run();
         INFO("Number of components ", cc.numberOfComponents());
         DEBUG("Estimating size of the largest component");
-        std::map<count, count> sizes = cc.getComponentSizes();
-        count largest_comp_size = 0;
-        for(auto it = sizes.cbegin(); it != sizes.cend(); ++it) {
-            DEBUG(it->second);
-            if (it->second > largest_comp_size) {
-                largest_comp_size = it->second;
-            }
-        }
-        INFO("Largest component size: ", largest_comp_size);
-        vd = largest_comp_size;
+        const auto compSizes = cc.getComponentSizes();
+        vd = std::max_element(
+                 compSizes.begin(), compSizes.end(),
+                 [](const auto &c1, const auto &c2) -> bool { return c1.second < c2.second; })
+                 ->second;
```

Comment:
This is UB if compSizes is empty.

Reply:
If the graph is not empty (there is a check before) then `compSizes` cannot be empty right?

Reply:
Yes, makes sense.